core/src/main/java/io/github/tcdl/msb/api/message/MetaMessage.java

```diff
@@ -40,8 +43,9 @@ public Builder(Integer ttl, Instant createdAt, ServiceDetails serviceDetails, Cl
         }     
 
         public MetaMessage build() {
-            Long durationMs = clock.instant().toEpochMilli() - this.createdAt.toEpochMilli();
-            return new MetaMessage(ttl, createdAt, durationMs, serviceDetails);
+            publishedAt = clock.instant();
+            Long durationMs = publishedAt.toEpochMilli() - this.createdAt.toEpochMilli();
```

Comment:
This place seems to be ideal for [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html)